README.md

```diff
@@ -1,37 +1,107 @@
 
-# Building
+## Requirements
 
-STLab is a standard CMake project. See the [running CMake](https://cmake.org/runningcmake) tutorial for an introduction to this tool.
+- A standards-compliant C++14, C++17, or C++20 compiler
+- **Use with a compiler in C++14-compliant mode** requires Boost.Optional and Boost.Variant >= 1.74.0
+- **Building** the library requires CMake 3.23 or later
+- **Testing or developing** the library requires Boost.Test >= 1.74.0
+
+## Building
+
+STLab is a standard CMake project. See the [running CMake](https://cmake.org/runningcmake) tutorial
+for an introduction to this tool.
+
+### Preparation
+
+1. Create a build directory outside this library's source tree.  In this guide, we'll use a sibling
+   directory called `BUILD`.
+
+2. If you are using the library in C++14-compliant mode or need to run the test suite, be sure you
+   have the necessary parts of Boost >= 1.74.0 installed.  Linux distributions usually make a
+   suitable version available through standard package managers.  On macOS or Linux, you can easilly
+   install Boost using [Homebrew](https://brew.sh/).  To install Boost on Windows, you can use
+   Boost's [binary installers](https://sourceforge.net/projects/boost/files/boost-binaries/).
+
+3. Install a version of CMake >= 3.23.  If you are on Debian or Ubuntu Linux you may need to use
+   `snap` to find one that's new enough.
+
+4. If you are using MSVC, you may need to set environment variables appropriately for your target
+   architecture by invoking `VCVARSALL.BAT` with an appropriate option.
+
+### Configure
+
+Run CMake in the root directory of this project, setting `../BUILD` as your build directory.  The
+basis of your command will be
+
+```
+cmake -S . -B ../BUILD
+```
+
+but there are other options you may need to append in order to be successful.  Among them:
+
+* `-DCMAKE_CXX_STANDARD=`[`14`|`17`|`20`|`23`] to build with compliance to the given C++ standard.
+* `-DSTLAB_USE_BOOST_CPP17_SHIMS=ON` if you are building in C++14 compliant mode.
+* `-DCMAKE_BUILD_TYPE=`[`Release`|`Debug`] to build the given configuration.
+* `-DBUILD_TESTING=OFF` if you only intend to build, but not test, this library.
+* `-DSTLAB_NO_STD_COROUTINES=TRUE` if you are targeting an Apple platform.
+* `-DBoost_USE_STATIC_LIBS=TRUE` if you will be testing on Windows.
+
+We also suggest the installation of [Ninja](https://ninja-build.org/) and its use by adding
+`-GNinja` to your cmake command line… but ninja is not required.
+
+A typical invocation might look like this:
+
+```
+cmake -S . -B ../BUILD -GNinja -DCMAKE_CXX_STANDARD=17 -DBUILD_TESTING=OFF
+```
+
+### Build
+
+If your configuration command was successful, go to your build directory (`cd ../BUILD`) and invoke:
+
+```
+cmake --build .
+```
+
+## Testing
+
+Running the tests in the `BUILD` directory is as simple as invoking 
+
+```
+ctest -C Debug
+```
+or
+```
+ctest -C Release
+```
+
+depending on which configuration (`CMAKE_BUILD_TYPE`) you chose to build.
```

Comment:
It should be mentioned that the `-C` option only makes sense if someone is not using a multi-config generator.

Reply:
Do you mean, "*is* using?"  because it seems to work regardless.  I'd rather keep complexity out if possible.

Reply:
Yes, I mean *is* using.

The documentation for this option doesn't specify what happens when a single-config generator is used with this option.

```
       -C <cfg>, --build-config <cfg>
              Choose configuration to test.

              Some  CMake-generated build trees can have multiple build configurations in the same tree.  This option
              can be used to specify which one should be tested.  Example configurations are Debug and Release.
```

I suppose if it is silently ignored, CMake will never change that unspecified behavior because it would break too many people. We can depend on this through social contract at least :). It makes me nervous that we're depending on unspecified behavior, but, yeah, I think we can, in practice, depend on it. I don't feel strongly about this suggestion.

Reply:
Counting this as resolved.